src/plugins/intel_gpu/src/graph/include/generic_layer.hpp

```diff
@@ -37,11 +37,15 @@ struct generic_layer : public primitive_base<generic_layer> {
     const kernel_selector::generic_kernel_params generic_params;
 
     size_t hash() const override {
-        size_t seed = primitive::hash();
+        size_t seed = 0;
         seed = hash_combine(seed, id);
         return seed;
     }
 
+    bool operator==(const primitive& rhs) const override {
+        return (id == rhs.id);
+    }
```

Comment:
It was already using id as hash. Why should we change this?

Reply:
Modify code to use generic_params instead of id.